gnovm/stdlibs/std/params.go

```diff
@@ -0,0 +1,66 @@
+package std
+
+import (
+	"fmt"
+	"unicode"
+
+	gno "github.com/gnolang/gno/gnovm/pkg/gnolang"
+)
+
+// ParamsInterface is the interface through which Gno is capable of accessing
+// the blockchain's params.
+//
+// The name is what it is to avoid a collision with Gno's Params, when
+// transpiling.
+type ParamsInterface interface {
+	SetString(key, val string)
+	SetBool(key string, val bool)
+	SetInt64(key string, val int64)
+	SetUint64(key string, val uint64)
+	SetBytes(key string, val []byte)
+}
+
+func X_setParamString(m *gno.Machine, key, val string) {
+	pk := pkey(m, key, "string")
+	GetContext(m).Params.SetString(pk, val)
+}
+
+func X_setParamBool(m *gno.Machine, key string, val bool) {
+	pk := pkey(m, key, "bool")
+	GetContext(m).Params.SetBool(pk, val)
+}
+
+func X_setParamInt64(m *gno.Machine, key string, val int64) {
+	pk := pkey(m, key, "int64")
+	GetContext(m).Params.SetInt64(pk, val)
+}
+
+func X_setParamUint64(m *gno.Machine, key string, val uint64) {
+	pk := pkey(m, key, "uint64")
+	GetContext(m).Params.SetUint64(pk, val)
+}
+
+func X_setParamBytes(m *gno.Machine, key string, val []byte) {
+	pk := pkey(m, key, "bytes")
+	GetContext(m).Params.SetBytes(pk, val)
+}
+
+func pkey(m *gno.Machine, key string, kind string) string {
+	// validate key.
+	if len(key) == 0 {
+		panic("empty param key")
+	}
+	first := rune(key[0])
+	if !unicode.IsLetter(first) && first != '_' {
+		panic("invalid param key: " + key)
+	}
+	for _, char := range key[1:] {
+		if !unicode.IsLetter(char) && !unicode.IsDigit(char) && char != '_' {
+			panic("invalid param key: " + key)
+		}
+	}
+
+	// decorate key with realm and type.
+	_, rlmPath := currentRealm(m)
+	return fmt.Sprintf("%s.%s.%s", rlmPath, key, kind)
```

Comment:
so you could have two parameters with the same key but different types?

Reply:
Yes, we require specifying the type in the key name, so it's essentially a "typed name." 

The reasoning is that if you create "foo" as an integer and others begin to use that value, changing the type of "foo" to a string later on could break dependencies. Therefore, we prefer the explicit approach.

Reply:
a bit weird but i can get onboard. wdyt if we require to explicitly put the suffix all the way down to the usage in the GnoVM? (so the public API of `SetParam` requires putting the suffix `.bool`, `.string`...)

Reply:
That's already the case; tm2 will panic if you don't follow this rule.

Reply:
Yeah I'm just saying what if instead of "building" it here we require to set it properly in Gno as well: `std.SetParam("mykey.bool", true)` => sets `gno.land/r/demo/morgan.mykey.bool`.

Reply:
oh yes, i forgot to adapt this part yesterday; here it is: 4a1c0bbbd

thx